packages/eslint-plugin/src/rules/dependency-checks.ts

```diff
@@ -0,0 +1,325 @@
+      obsoleteDependency: `The "{{packageName}}" is found in the "package.json" but is not used.`,
+      versionMismatch: `The "{{packageName}}" is found in the "package.json" but it's range is not matching the installed version. Installed version: {{version}}.`,
+      missingDependencySection: `The "dependencies" section is missing from the "package.json".`,
+    },
+  },
+  defaultOptions: [
+    {
+      buildTargets: ['build'],
+      checkMissingDependencies: true,
+      checkObsoleteDependencies: true,
+      checkVersionMismatches: true,
+      checkMissingPackageJson: true,
+      ignoredDependencies: [],
+    },
+  ],
+  create(
+    context,
+    [
+      {
+        buildTargets,
+        ignoredDependencies,
+        checkMissingDependencies,
+        checkObsoleteDependencies,
+        checkVersionMismatches,
+        checkMissingPackageJson,
+      },
+    ]
+  ) {
+    if (!(context.parserServices as any).isJSON) {
+      return {};
+    }
+    const fileName = normalizePath(context.getFilename());
+    // support only package.json and project.json
+    if (
+      !fileName.endsWith('/package.json') &&
+      !fileName.endsWith('/project.json')
+    ) {
+      return {};
+    }
+
+    const projectPath = normalizePath(
+      (global as any).projectPath || workspaceRoot
+    );
+    const sourceFilePath = getSourceFilePath(fileName, projectPath);
+    const { projectGraph, projectRootMappings, projectFileMap } =
+      readProjectGraph(RULE_NAME);
+
+    if (!projectGraph) {
+      return {};
+    }
+
+    const sourceProject = findProject(
+      projectGraph,
+      projectRootMappings,
+      sourceFilePath
+    );
+
+    // check if source project exists, it's library and has a build target
+    if (!sourceProject || sourceProject.type !== 'lib') {
+      return {};
+    }
+
+    // check if library has a build target
+    const buildTarget = buildTargets.find(
+      (t) => sourceProject.data.targets?.[t]
+    );
+    if (!buildTarget) {
+      return {};
+    }
+
+    // gather helper dependencies for @nx/js executors
+    const helperDependencies = getHelperDependenciesFromProjectGraph(
+      workspaceRoot,
+      sourceProject.name,
+      projectGraph
+    );
+
+    // find all dependencies for the project
+    const npmDeps = findProjectsNpmDependencies(
+      sourceProject,
+      projectGraph,
+      buildTarget,
+      {
+        helperDependencies: helperDependencies.map((dep) => dep.target),
+      },
+      removePackageJsonFromFileMap(projectFileMap)
+    );
+    const projDependencies = {
+      ...npmDeps.dependencies,
+      ...npmDeps.peerDependencies,
+    };
+    const projDependencyNames = Object.keys(projDependencies);
+
+    const projPackageJsonPath = join(
+      workspaceRoot,
+      sourceProject.data.root,
+      'package.json'
+    );
+    let projPackageJsonDeps: Record<string, string> = {};
+    if (existsSync(projPackageJsonPath)) {
```

Comment:
I would move the existence check into the `getAllDependencies` function so that this check isn't needed by the caller. The root `package.json` isn't guaranteed to be there either (although that's a super edge case).